pkg/graph/resolvers/system_intake_actions.go

```diff
@@ -236,3 +236,222 @@ func CreateSystemIntakeActionRequestEdits(
+		var errUpdateIntake error // declare this separately because if we use := on next line, compiler thinks we're declaring a new updatedIntake variable as well
+		updatedIntake, errUpdateIntake = store.UpdateSystemIntake(ctx, intake)
+		if errUpdateIntake != nil {
+			return errUpdateIntake
+		}
+
+		return nil
+	})
+
+	// save action (including additional info for email, if any)
+	errGroup.Go(func() error {
+		action := models.Action{
+			IntakeID:       &input.SystemIntakeID,
+			ActionType:     models.ActionTypeREJECT,
+			ActorName:      adminUserInfo.CommonName,
+			ActorEmail:     adminUserInfo.Email,
+			ActorEUAUserID: adminEUAID,
+			Step:           &intake.Step,
+		}
+		if input.AdditionalInfo != nil {
+			action.Feedback = null.StringFromPtr(input.AdditionalInfo)
+		}
+
+		_, errCreatingAction := store.CreateAction(ctx, &action)
+		if errCreatingAction != nil {
+			return errCreatingAction
+		}
+
+		return nil
+	})
+
+	// save admin note
+	if input.AdminNote != nil {
+		errGroup.Go(func() error {
+			adminNote := &models.SystemIntakeNote{
+				SystemIntakeID: input.SystemIntakeID,
+				AuthorEUAID:    adminEUAID,
+				AuthorName:     null.StringFrom(adminUserInfo.CommonName),
+				Content:        null.StringFromPtr(input.AdminNote),
+			}
+
+			_, errCreateNote := store.CreateSystemIntakeNote(ctx, adminNote)
+			if errCreateNote != nil {
+				return errCreateNote
+			}
+
+			return nil
+		})
+	}
+
+	if err := errGroup.Wait(); err != nil {
+		return nil, err
+	}
+
+	return updatedIntake, nil
+}
+
+// IssueLCID handles a Not Approved by GRB action on an intake as part of Admin Actions v2
+func IssueLCID(
+	ctx context.Context,
+	store *storage.Store,
+	fetchUserInfo func(context.Context, string) (*models.UserInfo, error),
+	input model.SystemIntakeIssueLCIDInput,
+) (*models.SystemIntake, error) {
+	adminEUAID := appcontext.Principal(ctx).ID()
+
+	adminUserInfo, err := fetchUserInfo(ctx, adminEUAID)
+	if err != nil {
+		return nil, err
+	}
+
+	intake, err := store.FetchSystemIntakeByID(ctx, input.SystemIntakeID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = lcidactions.IsIssueLCIDValid(intake)
+	if err != nil {
+		return nil, err
+	}
+
+	// if a LCID wasn't passed in, we generate one
+	newLCID, err := lcidactions.GenerateNewLCID(ctx, store, input.Lcid)
+	if err != nil {
+		return nil, err
+	}
+
+	// update workflow state
+	intake.Step = models.SystemIntakeStepDECISION
+	intake.State = models.SystemIntakeStateCLOSED
+	intake.DecisionState = models.SIDSLcidIssued
+
+	// update LCID-related fields
+	intake.LifecycleID = null.StringFrom(newLCID)
+	intake.LifecycleExpiresAt = &input.ExpiresAt
+	intake.LifecycleScope = null.StringFrom(input.Scope)
+	intake.DecisionNextSteps = null.StringFrom(input.NextSteps)
+	intake.TRBFollowUpRecommendation = &input.TrbFollowUp
+	intake.LifecycleCostBaseline = null.StringFromPtr(input.CostBaseline)
+
+	// update other fields, including IT Gov v1 workflow status
+	updatedTime := time.Now()
```

Comment:
Should I have been timestamping the intakes in the other actions!?

Reply:
[As discussed on Slack](https://cmsgov.slack.com/archives/CNU2B59UH/p1693345827726359?thread_ts=1693342307.551099&cid=CNU2B59UH) - just make sure to add it going forward. This PR adds it for the other v2 actions already in `main`, as well as the actions introduced in this PR.